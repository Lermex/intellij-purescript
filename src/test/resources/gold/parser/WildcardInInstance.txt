Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Main')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Prelude')
          PsiElement(whitespace)('\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
          PsiElement(whitespace)('\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(ImportModuleName)
          PSASTWrapperElement(Qualified)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
            PsiElement(proper name)('Console')
          PsiElement(whitespace)('\n\n-- Until the functional dependency gets added to purescript-eff, \n-- we need this here.\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('Monad')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('m')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          PsiElement(<=)('<=')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('MonadEff')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('eff')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('m')
          PsiElement(whitespace)(' ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(GenericIdentifier)
              PsiElement(identifier)('m')
              PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('eff')
                PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PSASTWrapperElement(TypeAnnotationName)
            PsiElement(identifier)('liftEff')
            PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(ForAll)
                PsiElement(forall)('forall')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(GenericIdentifier)
                  PsiElement(identifier)('a')
                PsiElement(.)('.')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(ConstrainedType)
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PSProperNameImpl(ProperName)
                            PsiElement(proper name)('Eff')
                            PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('eff')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(NamedBinder)
        <empty list>
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(GenericIdentifier)
          PsiElement(identifier)('monadEffEff')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSASTWrapperElement(Qualified)
            PSProperNameImpl(ProperName)
              PsiElement(proper name)('MonadEff')
              PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('eff')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PSProperNameImpl(ProperName)
                    PsiElement(proper name)('Eff')
                    PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('eff')
          PsiElement())(')')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('liftEff')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(NamedBinder)
            <empty list>
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)('\n\n-- This should generate a warning with the correct inferred type.\n')
      PSASTWrapperElement(TypeDeclaration)
        PSASTWrapperElement(TypeAnnotationName)
          PsiElement(identifier)('test')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('m')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('MonadEff')
                          PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(identifier)('_')
                    PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(GenericIdentifier)
                      PsiElement(identifier)('m')
                      PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(GenericIdentifier)
                        PsiElement(identifier)('m')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PSProperNameImpl(ProperName)
                            PsiElement(proper name)('Unit')
                            PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('liftEff')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('log')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Done"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PSASTWrapperElement(TypeAnnotationName)
          PsiElement(identifier)('test1')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PSProperNameImpl(ProperName)
                  PsiElement(proper name)('Eff')
                  PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PsiElement(identifier)('_')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PSProperNameImpl(ProperName)
                  PsiElement(proper name)('Unit')
                  PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test1')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('liftEff')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('log')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Done"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PSASTWrapperElement(TypeAnnotationName)
          PsiElement(identifier)('main')
          PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(GenericIdentifier)
                PsiElement(identifier)('eff')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('Eff')
                          PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(()('(')
                    PSASTWrapperElement(Row)
                      PSASTWrapperElement(GenericIdentifier)
                        PSIdentifierImpl(identifier)
                          PsiElement(identifier)('console')
                        PsiElement(whitespace)(' ')
                      PsiElement(::)('::')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PSProperNameImpl(ProperName)
                                PsiElement(proper name)('CONSOLE')
                                PsiElement(whitespace)(' ')
                      PsiElement(|)('|')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeVar)
                        PSASTWrapperElement(TypeConstructor)
                          <empty list>
                        PSASTWrapperElement(GenericIdentifier)
                          PsiElement(identifier)('eff')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PSProperNameImpl(ProperName)
                          PsiElement(proper name)('Unit')
                          PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(NamedBinder)
          <empty list>
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('test')