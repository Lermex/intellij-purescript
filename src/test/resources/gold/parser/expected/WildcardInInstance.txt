Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('Main')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Prelude')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ImportDeclaration)
        PsiElement(import)('import')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Control')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Monad')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Eff')
            PsiElement(.)('.')
          PSASTWrapperElement(ModuleName)
            PsiElement(proper name)('Console')
        PsiElement(whitespace)('\n\n-- Until the functional dependency gets added to purescript-eff, \n-- we need this here.\n')
      PSASTWrapperElement(TypeClassDeclaration)
        PsiElement(class)('class')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('Monad')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('m')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(Implies)
          PsiElement(operator)('<=')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(ClassName)
          PsiElement(proper name)('MonadEff')
          PsiElement(whitespace)(' ')
        PsiElement(identifier)('eff')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('m')
        PsiElement(whitespace)(' ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeVar)
              PsiElement(identifier)('m')
              PsiElement(whitespace)(' ')
          PsiElement(->)('->')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('eff')
                PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(TypeDeclaration)
          PsiElement(identifier)('liftEff')
          PsiElement(whitespace)(' ')
          PsiElement(::)('::')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(ForAll)
                PsiElement(forall)('forall')
                PsiElement(whitespace)(' ')
                PsiElement(identifier)('a')
                PsiElement(.)('.')
                PsiElement(whitespace)(' ')
                PSASTWrapperElement(ConstrainedType)
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('Eff')
                          PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('eff')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('a')
                        PsiElement(whitespace)(' ')
                    PsiElement(->)('->')
                    PsiElement(whitespace)(' ')
                    PSASTWrapperElement(Type)
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('m')
                          PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeAtom)
                        PSASTWrapperElement(TypeVar)
                          PsiElement(identifier)('a')
                          PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeInstanceDeclaration)
        PsiElement(instance)('instance')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('monadEffEff')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('MonadEff')
          PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeVar)
            PsiElement(identifier)('eff')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PsiElement(()('(')
          PSASTWrapperElement(Type)
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeConstructor)
                PSASTWrapperElement(Qualified)
                  PsiElement(proper name)('Eff')
                  PsiElement(whitespace)(' ')
            PSASTWrapperElement(TypeAtom)
              PSASTWrapperElement(TypeVar)
                PsiElement(identifier)('eff')
          PsiElement())(')')
          PsiElement(whitespace)(' ')
        PsiElement(where)('where')
        PsiElement(whitespace)('\n  ')
        PSASTWrapperElement(ValueDeclaration)
          PsiElement(identifier)('liftEff')
          PsiElement(whitespace)(' ')
          PsiElement(=)('=')
          PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('id')
                  PsiElement(whitespace)('\n\n-- This should generate a warning with the correct inferred type.\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('test')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('m')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('MonadEff')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(identifier)('_')
                    PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeVar)
                      PsiElement(identifier)('m')
                      PsiElement(whitespace)(' ')
                  PsiElement(=>)('=>')
                  PsiElement(whitespace)(' ')
                  PSASTWrapperElement(Type)
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('m')
                        PsiElement(whitespace)(' ')
                    PSASTWrapperElement(TypeAtom)
                      PSASTWrapperElement(TypeConstructor)
                        PSASTWrapperElement(Qualified)
                          PsiElement(proper name)('Unit')
                          PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('liftEff')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('log')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Done"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('test1')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PsiElement(proper name)('Eff')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PsiElement(identifier)('_')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(TypeConstructor)
              PSASTWrapperElement(Qualified)
                PsiElement(proper name)('Unit')
                PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('test1')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('liftEff')
                PsiElement(whitespace)(' ')
          PSASTWrapperElement(IdentInfix)
            PSASTWrapperElement(Qualified)
              PsiElement(operator)('$')
              PsiElement(whitespace)(' ')
          PSASTWrapperElement(Value)
            PSASTWrapperElement(PrefixValue)
              PSASTWrapperElement(Var)
                PSASTWrapperElement(Qualified)
                  PsiElement(identifier)('log')
                  PsiElement(whitespace)(' ')
              PSASTWrapperElement(StringLiteral)
                PsiElement(string)('"Done"')
                PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(TypeDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Type)
          PSASTWrapperElement(TypeAtom)
            PSASTWrapperElement(ForAll)
              PsiElement(forall)('forall')
              PsiElement(whitespace)(' ')
              PsiElement(identifier)('eff')
              PsiElement(.)('.')
              PsiElement(whitespace)(' ')
              PSASTWrapperElement(ConstrainedType)
                PSASTWrapperElement(Type)
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Eff')
                        PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PsiElement(()('(')
                    PSASTWrapperElement(Row)
                      PSASTWrapperElement(identifier)
                        PsiElement(identifier)('console')
                      PsiElement(whitespace)(' ')
                      PsiElement(::)('::')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(Type)
                        PSASTWrapperElement(TypeAtom)
                          PSASTWrapperElement(TypeConstructor)
                            PSASTWrapperElement(Qualified)
                              PsiElement(proper name)('CONSOLE')
                              PsiElement(whitespace)(' ')
                      PsiElement(|)('|')
                      PsiElement(whitespace)(' ')
                      PSASTWrapperElement(TypeVar)
                        PsiElement(identifier)('eff')
                    PsiElement())(')')
                    PsiElement(whitespace)(' ')
                  PSASTWrapperElement(TypeAtom)
                    PSASTWrapperElement(TypeConstructor)
                      PSASTWrapperElement(Qualified)
                        PsiElement(proper name)('Unit')
                        PsiElement(whitespace)('\n')
      PSASTWrapperElement(ValueDeclaration)
        PsiElement(identifier)('main')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(Value)
          PSASTWrapperElement(PrefixValue)
            PSASTWrapperElement(Var)
              PSASTWrapperElement(Qualified)
                PsiElement(identifier)('test')